pkgs/development/python-modules/pyqtgraph/default.nix

```
@@ -1,24 +1,49 @@
 { stdenv
+, pkgs
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
 , scipy
 , numpy
-, pyqt4
+, pyqt5
 , pyopengl
+, h5py
+, python
+, pytest
+, freefont_ttf
+, makeFontsConf
 }:
 
-buildPythonPackage rec {
+let
+  fontsConf = makeFontsConf {
+    fontDirectories = [ freefont_ttf ];
+  };
+in buildPythonPackage rec {
   pname = "pyqtgraph";
   version = "0.11.0";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "0p5k73wjfh0zzjvby8b5107cx7x0c2rdj66zh1nc8y95i0anf2na";
+  src = fetchFromGitHub {
+    owner = "pyqtgraph";
+    repo = "pyqtgraph";
+    rev = "pyqtgraph-${version}";
+    sha256 = "03fvpkqdn80ni51msvyivmghw41qk4vplwdqndkvzzzlppimdjbn";
   };
 
-  propagatedBuildInputs = [ scipy numpy pyqt4 pyopengl ];
+  propagatedBuildInputs = [ numpy pyqt5 scipy pyopengl h5py ];
+
+  checkInputs = [ pytest ];
+
+  checkPhase = with pkgs.qt5; ''
+    export QT_PLUGIN_PATH="${qtbase.bin}/${qtbase.qtPluginPrefix}"
+    export QT_QPA_PLATFORM=offscreen
+
```

```suggestion
    export DYLD_FRAMEWORK_PATH=/System/Library/Frameworks
```

Export that for darwin which should fix the tests.

---

Sorry it was a bit late. You should add an ``stdenv.lib.optionals stdenv.isDarwin`` to not export the var on every platform.